rdf-star-cg-spec.html

```diff
@@ -343,11 +345,78 @@ <h2>Parsing</h2>
+            <td>&quot;&lt;&lt;&quot; <a href="#grammar-production-nt-subject">subject</a> <a data-cite="N-TRIPLES#grammar-production-predicate">predicate</a> <a href="#grammar-production-nt-object">object</a> &quot;&gt;&gt;&quot;</td>
+          </tr>
+        </tbody>
+      </table>
+
+      <p class="note">As with Turtle*, the changes are that <a href="#grammar-production-nt-subject">`subject`</a> and <a href="#grammar-production-nt-object">`object`</a> productions have been extended to accept <a>embedded triples</a>, which are described by the new production <a href="#grammar-production-nt-embTriple">7</a>. N-Triples* does not include an <a href="#grammar-production-annotation">annotation</a> form.</p>
+    </section>
+
+    <section id="n-triples-star-parsing">
+      <h2>Parsing</h2>
+      <p>In contrast to [[N-TRIPLES]], N-Triples* allows recursion on the <a href="#grammar-production-nt-subject">`subject`</a> and <a href="#grammar-production-nt-object">`object`</a> productions.</p>
+
+    <p>An N-Triples* document defines an <a>RDF* graph</a> composed of a set of <a>RDF* triples</a>. The <a href="N-TRIPLES#grammar-production-triple">`triple`</a> production produces an <a>RDF* triple</a> defined by the terms constructed for <a href="#grammar-production-nt-subject">`subject`</a>, <a data-cite="N-TRIPLES#grammar-production-predicate">`predicate`</a> and <a href="#grammar-production-nt-object">`object`</a>.</p>
+
+      <p>In addition to the <a data-cite="N-TRIPLES#sec-parsing-terms">Term Constructors</a> defined in [[N-TRIPLES]], an additional constructor is defined for <a href="#grammar-production-nt-embTriple">`embTriple`</a> of type <a>RDF* triple</a> defined by the terms constructed for <a href="#grammar-production-nt-subject">`subject`</a>, <a data-cite="N-TRIPLES#grammar-production-predicate">`predicate`</a> and <a href="#grammar-production-nt-object">`object`</a>.</p>
+
+    <p>All other productions MUST be handled as specified by <a data-cite="N-TRIPLES#sec-parsing-terms">Section 8.1 of the N-Triples specification</a> [[N-TRIPLES]], while still applying the changes above recursively.</p>
+    </section>
+    </section>
+
+    <section id="n-triples-star">
+    <h2>N-Quads*</h2>
+
+    <p>The [[N-QUADS]] format is extended to describe the N-Quads* format using the same production updates described in the <a href="#n-triples-star-grammar">N-Triples* Grammar</a>.</p>
+
+    <p class="note">As RDF* describes <a>embedded triples</a> and not embedded quads, the <a data-cite="N-QUADS#grammar-production-graphLabel">`graphLabel`</a> component of an N-Quads <a data-cite="N-QUADS#grammar-production-statement">`statement`</a> does not apply to the <a href="#grammar-production-nt-embTriple">`embTriple`</a> component.</p>
+
+    <p>An N-Quads* document defines an <a>RDF* dataset</a> composed of
+      a single <a>default graph</a>, which is an <a>RDF* graph</a>,
+      and zero or more <a>named graphs</a>, which are <a>RDF graphs</a>.</p>
```

Comment:
```suggestion
      a single <a>default graph</a>, and zero or more <a>named graphs</a>,
      all of which are <a>RDF* graphs</a>.</p>
```

Reply:
> I’d rather not hold back the PR. 

+1

> Perhaps we should (temporarily) just eliminate the N-Quads* section until this has been properly discussed.

If you feel more comfortable with that, ok, but...

> it seems like a change with a number of implications for the semantics.

... I disagree. We don't need to invoke the semantics for that discussion, only the abstract syntax, which is what an N-Quads* parser is producing.

Rephrasing what @hartig wrote above:
```
<< <http://ex.com/a> <http://ex.com/b> <http://ex.com/c> >> <http://ex.com/pp> "oo" <http://ex.com/graph> .
```
The dataset serialized in this example has an empty default graph, and one named graph containing one (asserted) triple. The subject of this asserted triple is an embedded triple. That latter points suffices to make the named graph an RDF* graph.

> there is no syntax for creating triples in anything other than the default graph

"Creating" triples is ambiguous here. Let's be more precise. With the current grammar:

* triples can be *asserted* in either the default graph or named graphs;
* triples can be *embedded* in any quad, hence be in the constituents of either the default graph or named graphs.

In the abstract syntax, those are the only two ways I can see for a triple to be "in" a graph.